app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\User;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+
+class CampaignRegistrationController extends Controller
+{
```

Comment:
we're going to want to add a `create(Request $request) {...}` function + route here, so we can show a form that will allow a user to choose what character will be joining the campaign or if they'll be joining as a GM. The Create route should receive all the logged in user's characters. Also you should pass along the `code` as prop to the view that you create to help the front-ends